flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferManager.java

```diff
@@ -65,14 +65,10 @@
 	@GuardedBy("bufferQueue")
 	private int numRequiredBuffers;
 
-	public BufferManager(
-		MemorySegmentProvider globalPool,
-		InputChannel inputChannel,
-		int numRequiredBuffers) {
-
-		this.globalPool = checkNotNull(globalPool);
-		this.inputChannel = checkNotNull(inputChannel);
+	public BufferManager(InputChannel inputChannel, int numRequiredBuffers) {
 		checkArgument(numRequiredBuffers >= 0);
+		this.inputChannel = checkNotNull(inputChannel);
+		this.globalPool = inputChannel.inputGate.getMemorySegmentProvider();
```

Comment:
Actually I injected it in the constructor in previous version, but since we can not get ride of `InputChannel` completed in `BufferManager` as above said, then I reduce this unnecessary argument in constructor.

Reply:
I think it would be better to limit the reliance on this faulty cyclic dependency. That way it will be easier to get rid of it in the future.

Reply:
Makes sense